starter-code/src/functions-and-arrays.js

```diff
@@ -74,6 +186,24 @@ var wordsCount = [
   'disobedience',
   'matter'
 ];
+
+var searchResult = 0;
+
+var wordTried = "";
+
+function howManyTimes (wordsCount, wordTried){
+
+  for (i=0; i<wordsCount.length; i++){
+
+    if (wordTried === wordsCount[i]){ // returns true if the word we are looking is the only one on the array
+      searchResult += ;
+    }
+
+  }
+
+}
+
+
 // Bonus Quest
```

Comment:
And here is the correction for the matrix part : 
```
function greatestProduct(m) {
  var horizontal, vertical, diagonal;
  var greatProduct = 0;

  for (var i=0, nRows = m.length; i < nRows - 1; i++) {
    for (var j=0, nCols = m[0].length; j < nCols - 1; j++) {
      // 4 horizontal elements
      if (j < nCols-3) {
        horizontal = m[i][j] * m[i][j+1]   * m[i][j+2]   * m[i][j+3];
      }

      // 4 vertical elements
      if (i < nRows-3) {
        vertical   = m[i][j] * m[i+1][j]   * m[i+2][j]   * m[i+3][j];
      }

      // 4 Diagonal elements
      if (i < nRows-3 && j < nCols-3) {
        diagonal   = m[i][j] * m[i+1][j+1] * m[i+2][j+2] * m[i+3][j+3];
      }

      greatProduct = Math.max(greatProduct, horizontal, vertical, diagonal);
    }
  }

  return greatProduct;
}
```

But this one is quite tough!